provider/common/availabilityzones.go

```diff
@@ -204,21 +205,57 @@ func DistributeInstances(
 	return eligible, nil
 }
 
-// ValidateAvailabilityZone returns nil iff the availability
-// zone exists and is available, otherwise returns a NotValid
-// error.
+// LookupAvailabilityZone looks up an availability zone by name (or path) and
+// returns the zone object if found, nil and an error if not.
+func LookupAvailabilityZone(env ZonedEnviron, ctx context.ProviderCallContext, zones []AvailabilityZone, zone string) (AvailabilityZone, error) {
```

Comment:
Maybe SelectAvailabilityZone since we are not stricting looking anything up but selecting a match from a supplied list.

Also don't need the ctx as we're not making any cloud API calls.